### PR TITLE
Implement Ably Rule Resource with SQS Target

### PR DESCRIPTION
**Pull request by graham-russell**
This PR includes the following items
- tfplugindocs generated documentation
- Ably SQS rule resource with inbuilt source and target. 
- Updates to provider and models to represent Ably Rule for SQS target
- Playground tf templates updated
- Resource example updated for SQS integration rule
- Acceptance testing configured

### CODE REVIEW

**Comment by kavalerov** [comment]
Is this ready for review, or still draft?

File: internal/provider/resource_ably_rule_sqs.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+						Description: "JSON provides a text-based encoding",
+					},
+				}),
+			},
+		},
+	}, nil
+}
+
+func gen_plan_sqs_target_config(plan AblyRuleSqs, req_aws_auth ably_control_go.AwsAuthentication) ably_control_go.Target {
+
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply internal/provider/resource_ably_rule_sqs.go]: Tidied up

File: internal/provider/resource_ably_rule_sqs.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+	if v, ok := ably_rule.Target.(*ably_control_go.AwsSqsTarget); ok {
+		resp_target = AblyRuleTargetSqs{
+			Region:       v.Region,
+			AwsAccountID: v.AwsAccountID,
+			QueueName:    v.QueueName,
+			Enveloped:    v.Enveloped,
+			Format:       v.Format,
+		}
+		if a, ok := v.Authentication.Authentication.(*ably_control_go.AuthenticationModeCredentials); ok {
+
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply internal/provider/resource_ably_rule_sqs.go]: Tidied up

File: examples/resources/rule_aws_sqs.tf [hunk 3]
```diff
@@ -1,26 +1,19 @@
...
-
-resource "ably_rule_target_aws_sqs" "aws_sqs" {
-  region         = "eu-west-1"
-  aws_account_id = "123456789012"
-  queue_name     = "MyQueue"
-  authentication = {
-    authentication_mode = "credentials"
-    access_key_id       = "AKIAIOSFODNN7EXAMPLE"
-    secret_access_key   = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+resource "ably_rule_sqs" "rule0" {
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply examples/resources/rule_aws_sqs.tf:1]: Fixed

[the remark this answers is not in the document]
Comment (Morganamilo) [reply examples/resources/rule_aws_sqs.tf:1]: I didn't comment on every instance of this. There's still some messed up alignment in this PR.

**Review by kavalerov — APPROVED** [review]
Looks great, thanks. LGTM!

**Review by Morganamilo — APPROVED** [review]

**Pull request merged**